iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1179,4 +1179,28 @@ public boolean shouldOverwrite(org.apache.hadoop.hive.ql.metadata.Table mTable,
     }
     return COPY_ON_WRITE.equalsIgnoreCase(mode);
   }
+
+  @Override
+  public Boolean hasDeleteOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SnapshotContext since) {
+    TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
+    Table table = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
+    boolean foundSince = false;
+    for (Snapshot snapshot : table.snapshots()) {
+      if (!foundSince) {
+        if (snapshot.snapshotId() == since.getSnapshotId()) {
```

Comment:
is it sorted by snapshotId?

Reply:
No. It is sorted by `timestamp`

Reply:
that would be the same, isn't it? if the older timestamps could have a lower snapshot id above logic could be problematic.